docker-entrypoint.sh

```diff
@@ -126,19 +126,22 @@ if [ "$1" = 'rabbitmq-server' ]; then
 			].
 		EOF
 	fi
+fi
 
-	if [ "$ssl" ]; then
-		# Create combined cert
-		cat "$RABBITMQ_SSL_CERT_FILE" "$RABBITMQ_SSL_KEY_FILE" > /tmp/combined.pem
-		chmod 0400 /tmp/combined.pem
+if [ "$ssl" ]; then
+	# Create combined cert
+	# If calling this script with an exec this is problematic, as it needs to be owned 400 for perms but this code can run as root OR rabbitmq.
+	# As a workaround prefix the combined cert with the username so each user only operates on the file they control.
+	combined_file="/tmp/$(id -un)_combined.pem"
```

Comment:
`gosu` just allows us to drop from the root user to another user and group.  We start as root so that we can chown `/var/lib/rabbitmq` to make it easy on a bind mounted volume; for users on Mac/Windows that are mounting a directory from their host, this chown fails so it is useful for them to just run directly as the owner of that directory.